lib/algolia.ex

```diff
@@ -491,6 +491,186 @@ defmodule Algolia do
     |> inject_index_into_response(index)
   end
 
+  @doc """
+  Get all the synonyms of a index.
+
+  Only the index is required
+
+  ## Examples
+
+      iex> Algolia.export_synonyms("index")
+      {:ok, %{
+        "indexName" => "index",
+        "nbHits" => 33,
+        "hits" => [%{
+          "objectID" => "1539816124475_0",
+          "synonyms" => ["big", "large"],
+          "type" => "synonym"
+          "_highlightResult" => %{...}
+        }, ...]
+      }}
+  """
+  def export_synonyms(index, hits_per_page \\ 100) do
+    get_all_paginated_hits(index, &search_synonyms/3, hits_per_page)
+  end
+
+  defp get_all_paginated_hits(index, search, hits_per_page \\ 100) do
+    0
+    |> Stream.iterate(&(&1 + 1))
+    |> Stream.flat_map(fn page -> get_page_hits(index, page, hits_per_page, search) end)
+    |> Stream.take_while(&(&1 != :stop))
+  end
+
+  defp get_page_hits(index, page, hits_per_page, search) do
+    case search.(index, "", page: page, hits_per_page: hits_per_page) do
+      {:ok, %{"hits" => hits, "nbPages" => pages}} when page + 1 < pages ->
+        page_hits(hits)
+
+      {:ok, %{"hits" => hits, "nbHits" => nbHits}} when page + 1 < nbHits / hits_per_page ->
```

Comment:
Is this an edge case when Algolia doesn't return `nbPages` in their response JSON?